gemfirexd/tools/src/dunit/java/com/pivotal/gemfirexd/ddl/BugsDUnit.java

```diff
@@ -4660,6 +4666,60 @@ public void run() {
     }
   }
 
+  public void test41995() throws Exception {
+//    reduceLogLevelForTest("fine");
+    startVMs(1, 2);
+    VM serverVM1 = serverVMs.get(0);
+    VM serverVM2 = serverVMs.get(1);
+    Connection conn = TestUtil.getConnection();
+    Statement stmt = conn.createStatement();
+
+    stmt.execute("CREATE TABLE T1(COL1 INT, COL2 INT)");
+
+    PreparedStatement ps = conn.prepareStatement("INSERT INTO T1 VALUES (?, ?)");
+    for (int i = 1; i <= 100; i++) {
+      ps.setInt(1, i);
+      ps.setInt(2, i);
+      ps.execute();
+    }
+
+    try {
+      serverVM1.invoke(this.getClass(), "raiseCriticalUp");
+      serverVM2.invoke(this.getClass(), "raiseCriticalUp");
+      stmt.execute("delete from t1 where col1 > 0");
+    } finally {
```

Comment:
Had run the test without patch couple of times without patch and found that CRITICAL_UP was received and delete got cancelled. May be will run it a few more times will add some delay if required. I did notice that CRITICAL_UP did not go away immediately even after resetting the resource manager threshold.